testing/internal/e2e/vault/vault.go

```diff
@@ -286,6 +286,14 @@ func CreateKvPasswordCredential(t testing.TB, secretPath string, user string) (s
 
 	policyName = WritePolicy(t, t.Context(), policyFilePath)
 
+	// Use provided password or generate random one
+	if len(providedPassword) > 0 && providedPassword[0] != "" {
+		password = providedPassword[0]
+	} else {
+		password, err = base62.Random(16)
+		require.NoError(t, err)
+	}
```

Comment:
Did this work when you tested this?

It looks like `password` just gets overwritten below
```
// Create secret
password, err = base62.Random(16)
require.NoError(t, err)
```